### PR TITLE
[FLINK-11634][docs] Translate "State Backends" page into Chinese

### PR DESCRIPTION
**Pull request by shenzhu**
## What is the purpose of the change

FLINK-11634: https://issues.apache.org/jira/browse/FLINK-11634
Translate "State Backends" page into Chinese

## Brief change log

Translate page: https://ci.apache.org/projects/flink/flink-docs-master/docs/ops/state/state_backends/

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by wuchong** [comment]
@Myasuka , could you help to review this?

**Comment by Myasuka** [comment]
Thanks for the reminder, I have asked @Zakelly for help.

**Comment by Zakelly** [comment]
Thanks @wuchong and @Myasuka , I would like to review this.

**Comment by shenzhu** [comment]
Hey @Zakelly , sorry to bother you...but could you please have a look at this PR when you have a moment? Thanks!

File: docs/content.zh/docs/ops/state/state_backends.md [hunk 1]
```diff
@@ -332,17 +301,18 @@ public class MyOptionsFactory implements ConfigurableRocksDBOptionsFactory {
 
 {{< top >}}
 
-## Migrating from Legacy Backends
+## 自旧版本迁移
+
+从 **Flink 1.13** 版本开始，社区改进了 state backend 的公开类，进而帮助用户更好理解本地状态存储和 checkpoint 存储的区分。
+这个变化并不会影响 state backend 和 checkpointing 过程的运行时实现和机制，仅仅是为了更好地传达设计意图。
+用户可以在没有任何 state 或者 consistency 失效的情况下使用新的 API。
```
Comment (Zakelly) [inline docs/content.zh/docs/ops/state/state_backends.md:308]:
```suggestion
用户可以将现有作业迁移到新的API，同时不会损失原有 state。
```

**Review by Zakelly — COMMENTED** [review]
Thanks @shenzhu for your work! Please take a look at my comments. Moreover, there are some left-over 'RocksDBStateBackend' in untouched lines, which could better be changed to 'RocksDB' or 'EmbeddedRocksDBStateBackend'. Please also take a look, thanks.

**Comment by Zakelly** [comment]
How about modify or simplify the description of this PR? Thanks.

**Comment by shenzhu** [comment]
> Thanks @shenzhu for your work! Please take a look at my comments. Moreover, there are some left-over 'RocksDBStateBackend' in untouched lines, which could better be changed to 'RocksDB' or 'EmbeddedRocksDBStateBackend'. Please also take a look, thanks.

Thanks for your detailed review! I will work on the commets😄

**Review by Zakelly — APPROVED** [review]
Thanks for the update! Overall it LGTM, +1.  @Myasuka  would you like to take a brief look and merge this?

**Review by Myasuka — APPROVED** [review]
LGTM, thanks for the work of @shenzhu and @Zakelly 
Merging...

**Pull request merged**